fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1160,12 +1202,20 @@ ACTOR Future<Void> applyMetadataToCommittedTransactions(CommitBatchContext* self
 void writeMutation(CommitBatchContext* self, int64_t tenantId, const MutationRef& mutation) {
 	static_assert(TenantInfo::INVALID_TENANT == ENCRYPT_INVALID_DOMAIN_ID);
 	if (!self->pProxyCommitData->isEncryptionEnabled || tenantId == TenantInfo::INVALID_TENANT) {
```

Comment:
The `isRawAccess` check is probably an incorrect one and I was using it for `CODE_PROBE` only. I think this code block should look something like this.
```
if (!self->pProxyCommitData->isEncryptionEnabled) {
    self->toCommit.writeTypedMessage(mutation);
} else {
    // handle raw access
    if (tenantId == TenantInfo::INVALID_TENANT) {
        tenantId = getTenantIdFromMutation(mutation);
    }
    self->toCommit.writeTypedMessage(
        EncryptedMutationMessage::encrypt(arena, self->cipherKeys, tenantId /*domainId*/, mutation));
}
```